### DISCUSSION TITLE
[Help] Trying to persist storeCart data using nextJS, my states(cart and product) return as undefined

### DISCUSSION THREAD

**Discussion by victornasar** [pmndrs/zustand#2232] [Q&A]
I've tried using a useStore Hook to make my server and client rendered html be the same. For the most part of my day, I've been getting errors stating my states are undefined as I think the server isn't registering the states inside of localStorage.

 I reseted my code back to when I added a useStore hook, and I used it across my pages and components that use the cart and 
products. 
<img width="473" alt="Screenshot 2023-12-07 at 2 11 52 PM" src="https://github.com/pmndrs/zustand/assets/44656132/b6218b95-a9a3-47dd-a53a-7f422eee13aa">
<img width="473" alt="Screenshot 2023-12-07 at 2 12 04 PM" src="https://github.com/pmndrs/zustand/assets/44656132/01343816-3a6b-44d0-9e36-9af11adac7ed">
<img width="473" alt="Screenshot 2023-12-07 at 2 10 39 PM" src="https://github.com/pmndrs/zustand/assets/44656132/184b589a-0ddc-4662-8137-f6a7e889ff9d">
<img width="627" alt="Screenshot 2023-12-07 at 2 12 16 PM" src="https://github.com/pmndrs/zustand/assets/44656132/cee2c5d3-5379-43d4-9f82-b03520bbb33e">
<img width="627" alt="Screenshot 2023-12-07 at 2 12 27 PM" src="https://github.com/pmndrs/zustand/assets/44656132/dc1eb555-839a-4881-837a-bff257a90344">

**Comment by charkour** [comment]
Hi @victornasar, have you had a chance to look at these [docs](https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md#hydration-and-asynchronous-storages)?
